### PR TITLE
Add password input in stories

### PR DESCRIPTION
**Pull request by oorestisime**
#### What does this PR do?
Adds a passwordinput in the storybook.
It came 2-3 times already the last month so maybe exposing one directly could help.

- Not entirely happy with my choice of icons. 

#### Where should the reviewer start?
storybook
#### What testing has been done on this PR?
storybook
#### How should this be manually tested?
storybook
#### Do the grommet docs need to be updated?
no
#### Should this PR be mentioned in the release notes?
no
#### Is this change backwards compatible or is it a breaking change?
compat

### CODE REVIEW

File: src/js/components/TextInput/textinput.stories.js [hunk 1]
```diff
@@ -28,6 +27,26 @@ class SimpleTextInput extends Component {
   }
 }
 
+const PasswordInput = ({ value, ...rest }) => {
+  const [inputValue, setValue] = useState(value);
```
Comment (ShimiSun) [inline src/js/components/TextInput/textinput.stories.js]: woot-woot! The first hook! 🎉

File: src/js/components/TextInput/textinput.stories.js [hunk 2]
```diff
@@ -28,6 +27,26 @@ class SimpleTextInput extends Component {
...
+  return (
+    <Box width="medium" margin="large" round="small" border>
+      <TextInput
+        plain
+        type={reveal ? 'text' : 'password'}
+        value={inputValue}
+        onChange={event => setValue(event.target.value)}
+        {...rest}
+      />
+      <Button
```
Comment (ShimiSun) [inline src/js/components/TextInput/textinput.stories.js]: The button is taking the full space and the focus looks awkward 
screen shot 2019-03-01 at 6 03 08 pm

Not a must, bet will be better to have it fixed.

File: src/js/components/TextInput/textinput.stories.js [hunk 3]
```diff
@@ -229,5 +248,6 @@ class CustomSuggestionsTextInput extends Component {
 
 storiesOf('TextInput', module)
   .add('Simple TextInput', () => <SimpleTextInput />)
+  .add('Password input', () => <PasswordInput />)
```
Comment (ShimiSun) [inline src/js/components/TextInput/textinput.stories.js]: good story to have, Thank you!

File: src/js/components/TextInput/textinput.stories.js (hunk 2 above)
Comment (oorestisime) [reply src/js/components/TextInput/textinput.stories.js]: oups forgot row and align.

**Comment by ShimiSun** [comment]
I agree with the note about the choice of icons, let's follow up on it in a separate issue https://github.com/grommet/grommet-icons/issues/100

**Pull request merged**